### PR TITLE
[MAINT] follow several recommendations from the Scientific Python Library Development Guide

### PR DESCRIPTION
**Pull request by Remi-Gau**
- apply several recommendations mentioned in 
https://github.com/SIMEXP/giga_connectome/issues/44#issuecomment-1731609079

### CODE REVIEW

**Comment by htwangtw** [comment]
okay adding those extra options has raised some issues and revealed some very hacky things I did for good - 

- `test_load_atlas_setting`, `test_generate_group_mask`: i manually overrided the templateflow home to load customised atlas. this is not a typical use case of templateflow 
- `test_calculate_intranetwork_correlation`: we used the niftimasker to flatten label atlas (3D); solvable
- `test_get_bids_images`: this is from a hack to force pybids to index fmriprep outputs, you are a better person to judge if i am doing the right thing here

**Comment by Remi-Gau** [comment]
Oh I thought I was breaking things... OK will probably move the problematic things in a separate PR so that the things that are 'trivial' in this one can go through

File: pyproject.toml [hunk 1]
```diff
@@ -74,6 +74,11 @@ exclude = "giga_connectome/_version.py"
...
 
 [tool.pytest.ini_options]
+minversion = "7"
+log_cli_level = "INFO"
+xfail_strict = true
+addopts = ["-ra", "--strict-config", "--strict-markers"]
 markers = [
     "smoke: smoke tests that will run on a downsampled real dataset (deselect with '-m \"not smoke\"')",
 ]
+filterwarnings = ["error"]
```
Comment (htwangtw) [inline pyproject.toml:84]: commenting this line out should do the trick for now.

**Review by htwangtw — COMMENTED** [review]
We can move the warnings into separate issues since they need fixing eventually

File: giga_connectome/utils.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def get_bids_images(
         root=bids_dir,
         database_path=bids_dir,
         validate=False,
-        derivatives=True,
+        derivatives=False,
```
Comment (Remi-Gau) [inline giga_connectome/utils.py:29]: > test_get_bids_images: this is from a hack to force pybids to index fmriprep outputs, you are a better person to judge if i am doing the right thing here

This should take care of this warning.

This is a flag to index things in the derivative and not to say that the dataset being index is a derivative.

**Review by htwangtw — APPROVED** [review]
LGTM thanks!

**Pull request merged**